src/js/airtable-list-builder.js

```diff
@@ -0,0 +1,286 @@
+                $filterSelect.append($filterOption);
+              }
+
+              // Add the filter to the page.
+              $filterGroup.append($filterSelect);
+              $('div#airtable-list-filters').append($filterGroup);
+            }
+          }
+
+          // Handle the changing of the filters.
+          // store filter for each group
+          var filters = {};
+          $('#airtable-list-filters select').on( 'change', function( event ) {
+
+            var $filter = $( event.currentTarget );
+            // get group key
+            var $filterGroup = $filter.parents('.airtable-list-filter-group');
+            var filterGroup = $filterGroup.attr('data-filter-group');
+
+            // set filter for group
+            filters[ filterGroup ] = ($filter.val() =='*') ? '' : '.' + filterGroup + '--' + $filter.val();
+            // combine filters
+            var filterValue = concatValues( filters );
+
+            // set filter for Isotope
+            $contentArea.isotope({ filter: filterValue });
+          });
+
+          // Set height to equal height
+          if (config.equalHeight) {
+            var largestHeight = 0;
+            $('.airtable-list-record-row').each(function( index ) {
+              if ($(this).height() > largestHeight) {
+                largestHeight = $(this).height();
+              }
+            });
+
+            $('.airtable-list-record-row').height(largestHeight);
+          }
+
+          // Load isotope
+          $contentArea.isotope({
+            itemSelector: ".airtable-list-record-row",
+            layoutMode: "fitRows",
+            fitRows: {
+              gutter: config.gutter,
+              equalheight: true
+            }
+          });
+
+          $contentArea.on("arrangeComplete", function(event, filteredItems) {
+            if (!filteredItems.length) {
+              $('#airtable-list-no-results').show();
+            }
+            else {
+              $('#airtable-list-no-results').hide();
+            }
+          });
+        }
+      });
+    }
+  }
+
+  // flatten object by concatting values
+  function concatValues( obj ) {
+    var value = '';
+    for ( var prop in obj ) {
+      value += obj[ prop ];
+    }
+    return value;
+  }
+
+  // Convert a string into a class friendly string.
+  function stringToCSSClass(inputString) {
+    // Remove special characters and spaces, then split the string into words
+    const words = inputString.replace(/[^a-zA-Z0-9]/g, ' ').split(/\s+/);
+
+    // Join the words with hyphens, then convert to lowercase
+    const classFriendlyString = words.join('-').toLowerCase();
+
+    return classFriendlyString;
+  }
+
+  // Replace the token in the template
+  function replaceToken(template, token, value) {
+    return template.replace('%' + token + '%', value);
+  }
+
+  // Apply the data to the template.
+  function processTemplate(config, templateKey, data) {
```

Comment:
This seems like you've done a lot of work. Ever looked at https://alpinejs.dev/essentials/templating to help with stuff like this?

Reply:
No, I haven't seen that before. Would be interesting to look at.